pkgs/development/libraries/json-glib/default.nix

```diff
@@ -14,25 +14,35 @@
 , gnome3
 }:
 
+let
+  isCross = stdenv.buildPlatform != stdenv.targetPlatform;
+in
 stdenv.mkDerivation rec {
   pname = "json-glib";
   version = "1.6.2";
 
-  outputs = [ "out" "dev" "devdoc" ];
+  outputs = [ "out" "dev" ] ++ lib.optionals (!isCross) [ "devdoc" ];
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
     sha256 = "092g2dyy1hhl0ix9kp33wcab0pg1qicnsv0cj5ms9g9qs336cgd3";
   };
 
+  strictDeps = true;
+
+  depsBuildBuild = [
+    pkg-config
```

Comment:
Not sure how `pkg-config`'s target figures in things but I would expect `nativeBuildInputs` to work here as well.